app/src/main/java/com/example/drawingactivity/HueView.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import com.example.drawingactivity.ColorConstants.hueMaxFloat
+
+@SuppressLint("ClickableViewAccessibility")
+class HueView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onHueChanged: ((Float) -> Unit)? = null
+    var onSliderClicked: ((Boolean) -> Unit)? = null
+    private var hueSlider = hueMaxFloat
+    private var currentHue = 0f
+    private var sliderBitmap: Bitmap? = null
+    private var dy = hueSlider
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            dy = ((mEvent.y / height) * hueMaxFloat).coerceIn(0f, hueMaxFloat)
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    onSliderClicked?.invoke(true)
+                    setHue(dy)
+                    pickHue()
+                    invalidate()
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    setHue(dy)
+                    pickHue()
```

Comment:
`pickHue` invalidates. `setHue` could invoke the lambda. There's a bit of redundancy here

Reply:
deleted pick Hue and consolidated in setHue